src/hb-ot-layout.cc

```diff
@@ -442,6 +575,19 @@ hb_ot_layout_table_select_script (hb_face_t      *face,
   return false;
 }
 
+
+/**
+ * hb_ot_layout_table_get_feature_tags:
+ * @face: #hb_face_t to work upon
+ * @table_tag: HB_OT_TAG_GSUB or HB_OT_TAG_GPOS
+ * @start_offset: The offset into the feature-tag array to start returning
```

Comment:
I used similar wording for other start_offsets; happy to edit if there's a proposed change, but would prefer to keep them consistent ... so if the others are off-base too that is important.

Reply:
My understanding is that, the `start_offset` is the first item to start fetching. So say the `GSUB` table has 23 features, but the client code does not know the count and to avoid allocating in the heap, it uses a stack allocated array of length, say 5, and in the first call it sets `start_offset` to zero and `feature_count` to 5, in the next call `start_offset` is 5, and so on until it fetches all the features.

We need also to document the return value.

Reply:
Okay; makes sense. So, like here: https://github.com/harfbuzz/harfbuzz/blob/fa3ebf845fcffa827600fc4ca9fdde3aaa99f4c5/src/hb-ot-math.cc#L168 (looks like the most detailed occurrence, based on my searching)

Reply:
Yes.

Reply:
OK; I propagated that wording style and also inserted (allow-none) and (array length=) annotations to the getter functions that use the same _offset mechanism.

Reply:
Latest commit also relocates the comment in hb_ot_layout_feature_get_characters regarding worrying about char_count's output and input being equal from the parameter block to a proper note, with a bit of rewording. I'm undecided if that ought to stay (or even be replicated elsewhere); I'm happy with whatever approach others prefer.